grails-app/services/com/unifina/service/SessionService.groovy

```diff
@@ -1,49 +1,51 @@
 package com.unifina.service
 
+import com.lambdaworks.redis.RedisAsyncConnection
 import com.lambdaworks.redis.RedisClient
 import com.lambdaworks.redis.RedisConnection
 import com.lambdaworks.redis.RedisURI
 import com.unifina.domain.security.SecUser
 import com.unifina.security.SessionToken
 import com.unifina.utils.MapTraversal
 import grails.util.Holders
+import org.joda.time.DateTime
 import org.springframework.util.Assert
 
 class SessionService {
 	static final int TOKEN_LENGTH = 64
 	static final int TTL_HOURS = 3
 
-	RedisClient redisClient
+	private RedisClient redisClient
+	private RedisAsyncConnection<String, String> connection
 
 	SessionToken generateToken(SecUser user) {
 		SessionToken sk = new SessionToken(TOKEN_LENGTH, user, TTL_HOURS)
-		RedisConnection<String, String> connection = getRedisClient().connect()
-		connection.setex(sk.getToken(), TTL_HOURS * 3600, sk.getUser().id.toString())
+		getConnection().setex(sk.getToken(), TTL_HOURS * 3600, sk.getUser().id.toString())
```

Comment:
Should we wait (block) for the request to complete? Otherwise the following order of events is possible:
- Token is created and Redis request is fired with the non-blocking async client
- Token is returned to user
- User attempts to use token, but it is not yet written to redis!